models/cable/R/write.config.CABLE.R

```diff
@@ -0,0 +1,125 @@
+write.config.CABLE <- function(defaults, trait.values, settings, run.id) {
+
+  # Please follow the PEcAn style guide:
+  # https://pecan.gitbooks.io/pecan-documentation/content/developers_guide/Coding_style.html
+  
+  # Note that `library()` calls should _never_ appear here; instead, put
+  # packages dependencies in the DESCRIPTION file, under "Imports:".
+  # Calls to dependent packages should use a double colon, e.g.
+  #    `packageName::functionName()`.
+  # Also, `require()` should be used only when a package dependency is truly
+  # optional. In this case, put the package name under "Suggests:" in DESCRIPTION. 
+  
+  # find out where to write run/ouput
+  rundir <- file.path(settings$host$rundir, run.id)
+  outdir <- file.path(settings$host$outdir, run.id)
+  
+  #-----------------------------------------------------------------------
+  # create launch script (which will create symlink)
+  if (!is.null(settings$model$jobtemplate) && file.exists(settings$model$jobtemplate)) {
+    jobsh <- readLines(con = settings$model$jobtemplate, n = -1)
+  } else {
+    jobsh <- readLines(con = system.file("template.job", package = "PEcAn.CABLE"), n = -1)
+  }
+  
+  # create host specific setttings
+  hostsetup <- ""
+  if (!is.null(settings$model$prerun)) {
+    hostsetup <- paste(hostsetup, sep = "\n", paste(settings$model$prerun, collapse = "\n"))
+  }
+  if (!is.null(settings$host$prerun)) {
+    hostsetup <- paste(hostsetup, sep = "\n", paste(settings$host$prerun, collapse = "\n"))
+  }
+  
+  hostteardown <- ""
+  if (!is.null(settings$model$postrun)) {
+    hostteardown <- paste(hostteardown, sep = "\n", paste(settings$model$postrun, collapse = "\n"))
+  }
+  if (!is.null(settings$host$postrun)) {
+    hostteardown <- paste(hostteardown, sep = "\n", paste(settings$host$postrun, collapse = "\n"))
+  }
+  
+  # create job.sh
+  jobsh <- gsub("@HOST_SETUP@", hostsetup, jobsh)
+  jobsh <- gsub("@HOST_TEARDOWN@", hostteardown, jobsh)
+  
+  jobsh <- gsub("@SITE_LAT@", settings$run$site$lat, jobsh)
+  jobsh <- gsub("@SITE_LON@", settings$run$site$lon, jobsh)
+  jobsh <- gsub("@SITE_MET@", settings$run$site$met, jobsh)
+  
+  jobsh <- gsub("@START_DATE@", settings$run$start.date, jobsh)
+  jobsh <- gsub("@END_DATE@", settings$run$end.date, jobsh)
+  
+  jobsh <- gsub("@OUTDIR@", outdir, jobsh)
+  jobsh <- gsub("@RUNDIR@", rundir, jobsh)
+  
+  jobsh <- gsub("@BINARY@", settings$model$binary, jobsh)
+  
+  writeLines(jobsh, con = file.path(settings$rundir, run.id, "job.sh"))
+  Sys.chmod(file.path(settings$rundir, run.id, "job.sh"))
+  
+  #-----------------------------------------------------------------------
+  ### Edit a templated config file for runs
+  if (!is.null(settings$model$config) && file.exists(settings$model$config)) {
+    config.text <- readLines(con = settings$model$config, n = -1)
+  } else {
+    filename <- system.file(settings$model$config, package = "PEcAn.CABLE")
```

Comment:
This will throw an error because you will only get to this step if ```settings$model$config``` is null. You should replace that settings bit with the name of the model-specific config file that you're editing (probably a Fortran Namelist) and the default, templated version of that file needs to be added to the pull request.